In [1]:
import numpy as np
import pandas as pd

# Dépenses alimentaires pour les vacances en décembre 2025

In [4]:
depenses_alimentation = pd.read_csv('depenses_alimentation.csv', index_col="Qui ?")

In [5]:
depenses_alimentation

alimentation_sans_loulou  alimentation_avec_loulou
Qui ?                                                     
Olivet                     11.00                    135.39
Olivet                    103.75                    103.75
Olivet                     20.00                    203.39
Olivet                     46.80                       NaN
Olivet                    264.40                       NaN
Olivet                     20.00                       NaN
Crans                      20.00                       NaN
Pau                       132.00                       NaN
Pau                         7.00                     45.00

In [6]:
depenses_alimentation_olivet = depenses_alimentation.loc['Olivet'].sum().sum()
print(f"Dépenses alimentation Olivet : {depenses_alimentation_olivet} €")

depenses_alimentation_crans = depenses_alimentation.loc['Crans'].sum().sum()
print(f"Dépenses alimentation Crans : {depenses_alimentation_crans} €")

depenses_alimentation_pau = depenses_alimentation.loc['Pau'].sum().sum()
print(f"Dépenses alimentation Pau : {depenses_alimentation_pau} €")

depenses_alimentation_chalon = 0
print(f"Dépenses alimentation Chalon : {depenses_alimentation_chalon} €")

Dépenses alimentation Olivet : 908.48 €
Dépenses alimentation Crans : 20.0 €
Dépenses alimentation Pau : 184.0 €
Dépenses alimentation Chalon : 0 €


NOTE : la ligne 1 (2 fois 103.75 : note de courses divisée par 2 - à voir si cela convient)

In [7]:
alimentation_avec_loulou = depenses_alimentation.alimentation_avec_loulou.sum()
print(f"Dépenses alimentation avec Loulou : {alimentation_avec_loulou} €")

Dépenses alimentation avec Loulou : 487.53 €


In [8]:
alimentation_sans_loulou = depenses_alimentation.alimentation_sans_loulou.sum()
print(f"Dépenses alimentation sans Loulou : {alimentation_sans_loulou} €")

Dépenses alimentation sans Loulou : 624.95 €


## Repartition des dépenses alimentaires

### Avec Loulou

In [9]:
total_adultes = 7
total_enfants = 7  # Rose n'est pas comptabilisée
coef_adulte = 1
coef_enfant = 1
coef_olivet = 2 * coef_enfant + 2 * coef_adulte
coef_crans = 1 * coef_enfant + 2 * coef_adulte
coef_pau = 2 * coef_enfant + 1 * coef_adulte
coef_chalon = 2 * coef_enfant + 2 * coef_adulte

In [10]:
assert coef_olivet + coef_crans + coef_pau + coef_chalon == (total_enfants * coef_enfant + total_adultes * coef_adulte)

In [11]:
alimentation_chalon = alimentation_avec_loulou * coef_chalon / (total_enfants * coef_enfant + total_adultes * coef_adulte)
alimentation_olivet_1 = alimentation_avec_loulou * coef_olivet / (total_enfants * coef_enfant + total_adultes * coef_adulte)
alimentation_crans_1 = alimentation_avec_loulou * coef_crans / (total_enfants * coef_enfant + total_adultes * coef_adulte)
alimentation_pau_1 = alimentation_avec_loulou * coef_pau / (total_enfants * coef_enfant + total_adultes * coef_adulte)

In [12]:
assert alimentation_chalon + alimentation_olivet_1 + alimentation_crans_1 + alimentation_pau_1 == alimentation_avec_loulou

### Sans Loulou

In [13]:
total_adultes = 5
total_enfants = 5  # Rose n'est pas comptabilisée
coef_adulte = 1
coef_enfant = 1
coef_olivet = 2 * coef_enfant + 2 * coef_adulte
coef_crans = 1 * coef_enfant + 2 * coef_adulte
coef_pau = 2 * coef_enfant + 1 * coef_adulte

assert coef_olivet + coef_crans + coef_pau == (total_enfants * coef_enfant + total_adultes * coef_adulte)

In [14]:
alimentation_olivet_2 = alimentation_sans_loulou * coef_olivet / (total_enfants * coef_enfant + total_adultes * coef_adulte)
alimentation_crans_2 = alimentation_sans_loulou * coef_crans / (total_enfants * coef_enfant + total_adultes * coef_adulte)
alimentation_pau_2 = alimentation_sans_loulou * coef_pau / (total_enfants * coef_enfant + total_adultes * coef_adulte)

assert np.isclose(alimentation_olivet_2 + alimentation_crans_2 + alimentation_pau_2, alimentation_sans_loulou)

In [15]:
print("Répartition des dépenses alimentaires :")
print(f"  Olivet : {alimentation_olivet_1 + alimentation_olivet_2:.2f} €")
print(f"  Crans : {alimentation_crans_1 + alimentation_crans_2:.2f} €")
print(f"  Pau : {alimentation_pau_1 + alimentation_pau_2:.2f} €")
print(f"  Chalon : {alimentation_chalon:.2f} €")

Répartition des dépenses alimentaires :
  Olivet : 389.27 €
  Crans : 291.96 €
  Pau : 291.96 €
  Chalon : 139.29 €


In [16]:
assert np.isclose(alimentation_chalon + alimentation_olivet_1 + alimentation_olivet_2 + alimentation_crans_1 + alimentation_crans_2 + alimentation_pau_1 + alimentation_pau_2, alimentation_avec_loulou + alimentation_sans_loulou)

# Dépenses draps et serviettes

In [17]:
depenses_drap_olivet = 40
depenses_drap_crans = 30  # ou 40 ?
depenses_drap_pau = 30
depenses_drap_chalon = 40

cout_total_draps = depenses_drap_olivet + depenses_drap_crans + depenses_drap_pau + depenses_drap_chalon

print("Répartition des dépenses de draps et serviettes :")
print(f"  Olivet : {depenses_drap_olivet} €")
print(f"  Crans : {depenses_drap_crans} €")
print(f"  Pau : {depenses_drap_pau} €")
print(f"  Chalon : {depenses_drap_chalon} €")

Répartition des dépenses de draps et serviettes :
  Olivet : 40 €
  Crans : 30 €
  Pau : 30 €
  Chalon : 40 €


# Dépenses logement et ménage

In [18]:
depenses_logement = 2100 + 280  # location + ménage
depenses_logement_par_nuit = depenses_logement / 7

In [19]:
## Répartition par nuitée avec Loulou

nombre_nuits = 3

total_adultes = 7
total_enfants = 8  # Rose est comptabilisée
coef_adulte = 1
coef_enfant = 0.5
coef_olivet = 2 * coef_enfant + 2 * coef_adulte
coef_crans = 2 * coef_enfant + 2 * coef_adulte
coef_pau = 2 * coef_enfant + 1 * coef_adulte
coef_chalon = 2 * coef_enfant + 2 * coef_adulte

assert coef_olivet + coef_crans + coef_pau + coef_chalon == (total_enfants * coef_enfant + total_adultes * coef_adulte)

In [20]:
logement_chalon = depenses_logement_par_nuit * coef_chalon / (total_enfants * coef_enfant + total_adultes * coef_adulte) * nombre_nuits
logement_olivet_1 = depenses_logement_par_nuit * coef_olivet / (total_enfants * coef_enfant + total_adultes * coef_adulte) * nombre_nuits
logement_crans_1 = depenses_logement_par_nuit * coef_crans / (total_enfants * coef_enfant + total_adultes * coef_adulte) * nombre_nuits   
logement_pau_1 = depenses_logement_par_nuit * coef_pau / (total_enfants * coef_enfant + total_adultes * coef_adulte) * nombre_nuits

assert np.isclose(logement_chalon + logement_olivet_1 + logement_crans_1 + logement_pau_1, depenses_logement_par_nuit * nombre_nuits)

In [21]:
## Répartition par nuitée sans Loulou

nombre_nuits = 4
total_adultes = 5
total_enfants = 6
coef_adulte = 1
coef_enfant = 0.5
coef_olivet = 2 * coef_enfant + 2 * coef_adulte
coef_crans = 2 * coef_enfant + 2 * coef_adulte
coef_pau = 2 * coef_enfant + 1 * coef_adulte

assert coef_olivet + coef_crans + coef_pau == (total_enfants * coef_enfant + total_adultes * coef_adulte)

In [22]:
logement_olivet_2 = depenses_logement_par_nuit * coef_olivet / (total_enfants * coef_enfant + total_adultes * coef_adulte) * nombre_nuits
logement_crans_2 = depenses_logement_par_nuit * coef_crans / (total_enfants * coef_enfant + total_adultes * coef_adulte) * nombre_nuits   
logement_pau_2 = depenses_logement_par_nuit * coef_pau / (total_enfants * coef_enfant + total_adultes * coef_adulte) * nombre_nuits

assert np.isclose(logement_olivet_2 + logement_crans_2 + logement_pau_2, depenses_logement_par_nuit * nombre_nuits)

In [23]:
print("Répartition des dépenses de logement :")
print(f"  Chalon : {logement_chalon:.2f} €")
print(f"  Olivet : {logement_olivet_1 + logement_olivet_2:.2f} €")
print(f"  Crans : {logement_crans_1 + logement_crans_2:.2f} €")
print(f"  Pau : {logement_pau_1 + logement_pau_2:.2f} €")

assert np.isclose(logement_chalon + logement_olivet_1 + logement_olivet_2 + logement_crans_1 + logement_crans_2 + logement_pau_1 + logement_pau_2, depenses_logement)

Répartition des dépenses de logement :
  Chalon : 278.18 €
  Olivet : 788.18 €
  Crans : 788.18 €
  Pau : 525.45 €


# Dépenses totales par foyer

In [24]:
cout_total_olivet = alimentation_olivet_1 + alimentation_olivet_2 + depenses_drap_olivet + logement_olivet_1 + logement_olivet_2
cout_total_crans = alimentation_crans_1 + alimentation_crans_2 + depenses_drap_crans + logement_crans_1 + logement_crans_2
cout_total_pau = alimentation_pau_1 + alimentation_pau_2 + depenses_drap_pau + logement_pau_1 + logement_pau_2
cout_total_chalon = alimentation_chalon + depenses_drap_chalon + logement_chalon

In [25]:
print(f"Coût Olivet : {cout_total_olivet:.2f} €")
print(f"Coût Crans : {cout_total_crans:.2f} €")
print(f"Coût Pau : {cout_total_pau:.2f} €")
print(f"Coût Chalon : {cout_total_chalon:.2f} €")

Coût Olivet : 1217.46 €
Coût Crans : 1110.14 €
Coût Pau : 847.41 €
Coût Chalon : 457.48 €


# Versement à faire par chacun pour équilibrer les dépenses

In [26]:
depenses_actuelles_crans = depenses_logement + cout_total_draps + depenses_alimentation_crans
print(f"Dépenses actuelles Crans : {depenses_actuelles_crans:.2f} €")

depenses_actuelles_olivet = depenses_alimentation_olivet
print(f"Dépenses actuelles Olivet : {depenses_actuelles_olivet:.2f} €")

depenses_actuelles_pau = depenses_alimentation_pau
print(f"Dépenses actuelles Pau : {depenses_actuelles_pau:.2f} €")

depenses_actuelles_chalon = depenses_alimentation_chalon
print(f"Dépenses actuelles Chalon : {depenses_actuelles_chalon:.2f} €")

Dépenses actuelles Crans : 2540.00 €
Dépenses actuelles Olivet : 908.48 €
Dépenses actuelles Pau : 184.00 €
Dépenses actuelles Chalon : 0.00 €


In [27]:
print(f"Olivet doit `{cout_total_olivet - depenses_actuelles_olivet:.2f} €` à Crans.")

Olivet doit `308.98 €` à Crans.


In [28]:
papier_cadeau = 3.80
print(f"Pau doit `{cout_total_pau + papier_cadeau - depenses_actuelles_pau:.2f} €` à Crans.")

Pau doit `667.21 €` à Crans.


In [29]:
print(f"Chalon doit `{cout_total_chalon - depenses_actuelles_chalon:.2f} €` à Crans.")

Chalon doit `457.48 €` à Crans.


In [31]:
# Verify calculations
assert np.isclose(308.98 + 667.21 + 457.48 + cout_total_crans - papier_cadeau, depenses_actuelles_crans)